jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JDTUtils.java

```diff
@@ -634,17 +635,24 @@ public static boolean hasLeadingSlash(String uriString) {
     /**
      * Returns a list of all accessors (getter and setter) of the given field.
      * 
-     * @param field
+     * @param unit, field
      * @return a list of accessor methods
      * @throws JavaModelException
      */
-    public static List<IMethod> getAccessors(IField field) throws JavaModelException {
-        List<IMethod> accessors = new ArrayList<>();
-        IMethod getter = GetterSetterUtil.getGetter(field);
-        IMethod setter = GetterSetterUtil.getSetter(field);
-        for (IMethod accessor : new IMethod[] { getter, setter }) {
-            if (accessor != null) {
-                accessors.add(accessor);
+    public static List<IMethod> getFieldAccessors(ICompilationUnit unit, IField field) throws JavaModelException {
+        List<IMethod> accessors = new ArrayList<IMethod>();
+        String fieldName = field.getElementName();
+        fieldName = fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
+        List<String> accessorNames = new ArrayList<String>();
+        accessorNames.add("get" + fieldName);
+        accessorNames.add("set" + fieldName);
+        accessorNames.add("is" + fieldName);
```

Comment:
"isField" is the accessor of a field only when the field is a boolean. We can add a check for that.

Reply:
Yes, but sometimes, even in the field is boolean the accessor still uses "getField()" instead of "isField()", therefore it might be safer to check for both for edge cases. What do you think? Or should we just check for "isField()" whenever the field is boolean?

Reply:
This is a good point. I did some testing and found that if the field is a boolean, only the method with name `isField` is identified as the getter (i.e. the expected `JsonbException` is thrown). `getField` is not identified as a getter so there is not any exception even if we have a JsonbTransient and another Jsonb annotation on it. 

So I think we should just check for "isField" when the field is boolean.